test/e2e/cluster_upgrade_runtimesdk.go

```diff
@@ -286,17 +271,19 @@ func extensionConfig(specName string, namespace *corev1.Namespace) *runtimev1.Ex
 			// when configuring external patches.
 			Name: specName,
 			Annotations: map[string]string{
-				runtimev1.InjectCAFromSecretAnnotation: fmt.Sprintf("%s/webhook-service-cert", namespace.Name),
+				runtimev1.InjectCAFromSecretAnnotation: "test-extension-system/test-extension-webhook-service-cert", // Note: this assumes the test extension get deployed in the default namespace defined in its own runtime-extensions-components.yaml
 			},
 		},
 		Spec: runtimev1.ExtensionConfigSpec{
 			ClientConfig: runtimev1.ClientConfig{
 				Service: &runtimev1.ServiceReference{
-					Name:      "webhook-service",
-					Namespace: namespace.Name,
+					Name:      "test-extension-webhook-service",
+					Namespace: "test-extension-system", // Note: this assumes the test extension get deployed in the default namespace defined in its own runtime-extensions-components.yaml
 				},
 			},
 			NamespaceSelector: &metav1.LabelSelector{
+				// Note: we are limiting the test extension to be used by the namespace where the test is run; this could
+				// become a limitation in future when there will be more than one test using this test extension / running in parallel.
```

Comment:
> come a limitation in future when there will be more than one test using this test extension / running in parallel.

Not sure if that's true. I would assume that every e2e test has to deploy its own ExtensionConfig. We should avoid leaking ExtensionConfigs between separate e2e tests

Reply:
Removed the controversial part; I'm not sure we support having more ExtensionConfig for the same RuntimeExtension, but it is not relevant now